File: src/test/java/samples/languages/java/models/Identity.java
package samples.languages.java.models;

import jakarta.validation.constraints.NotBlank;

public class Identity {
    
    private String id;
    
    @NotBlank
    private String category;
    
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }
}

File: src/test/scala/samples/languages/scala/v3/models/Identity.scala
package samples.languages.scala.v3.models

import jakarta.validation.constraints.NotBlank

import scala.beans.BeanProperty


abstract class Identity {

  @BeanProperty
  @NotBlank
  var id : String = _
  
  @BeanProperty
  @NotBlank
  val category : String

}
